lib/src/ui/puzzle/puzzle_dashboard_screen.dart

```diff
@@ -141,3 +147,80 @@ class _PuzzleButton extends StatelessWidget {
     );
   }
 }
+
+class DaysSelector extends ConsumerWidget {
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final session = ref.watch(userSessionStateProvider);
+    final day = ref.watch(daysProvider);
+    return session != null
+        ? InkWell(
+            onTap: () => showChoicesPicker(
+              context,
+              choices: Days.values,
+              selectedItem: day,
+              labelBuilder: (t) => Text(t.toString()),
+              onSelectedItemChanged: (newDay) {
+                ref.read(daysProvider.notifier).state = newDay;
+              },
+            ),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.end,
+              children: [
+                Text(day.toString()),
+                const Icon(Icons.arrow_drop_down),
+              ],
+            ),
+          )
+        : const SizedBox.shrink();
+  }
+}
+
+enum Days {
+  oneday,
+  twodays,
+  week,
+  twoweeks,
+  month,
+  twomonths,
+  threemonths;
+
+  @override
+  String toString() {
+    switch (this) {
+      case Days.oneday:
+        return '1 day';
```

Comment:
There's an available translation here: `context.l10n.nbDays`;
 You can either use a provider or a function that takes a `BuildContext`, see `puzzle_theme.dart` for examples.